pkg/storage/flamegraph_flat.go

```diff
@@ -86,32 +84,54 @@ func GenerateFlamegraphFlat(ctx context.Context, tracer trace.Tracer, metaStore
 			})
 
 			if index < len(cur.Children) && bytes.Equal(cur.Children[index].GetMeta().GetLocation().GetId(), nextID[:]) {
-				child = cur.Children[index]
+				// The node already exists in the flamegraph, we can simply add the value and diff value and continue with this child's children next.
```

Comment:
I think this still misses the case where we have two stack traces that have the same inlined location in the "middle" of the stack but different leafs. Because the second stack added will not add the cumulative to all of the functions in the flamegraph that belong to this location.

Reply:
Makes sense. I'll do that one as a follow-up if that's ok?

Reply:
Since this is behind a feature flag, I'm happy with that.